src/SearchEngine/Masa.BuildingBlocks.SearchEngine.AutoComplete/BaseAutoCompleteClient.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace Masa.BuildingBlocks.SearchEngine.AutoComplete;
+
+public abstract class BaseAutoCompleteClient : IAutoCompleteClient
+{
+    public virtual Task<GetResponse<AutoCompleteDocument<Guid>, Guid>> GetAsync(string keyword, AutoCompleteOptions? options = null,
+        CancellationToken cancellationToken = default)
+        => GetAsync<Guid>(keyword, options, cancellationToken);
+
+    public virtual Task<GetResponse<AutoCompleteDocument<TValue>, TValue>> GetAsync<TValue>(string keyword,
+        AutoCompleteOptions? options = null,
+        CancellationToken cancellationToken = default) where TValue : notnull
+        => GetAsync<AutoCompleteDocument<TValue>, TValue>(keyword, options, cancellationToken);
+
+    public abstract Task<GetResponse<TAudoCompleteDocument, TValue>> GetAsync<TAudoCompleteDocument, TValue>(string keyword,
+        AutoCompleteOptions? options = null,
+        CancellationToken cancellationToken = default) where TAudoCompleteDocument : AutoCompleteDocument<TValue> where TValue : notnull;
+
+    public virtual Task<SetResponse> SetAsync(AutoCompleteDocument<Guid> document, SetOptions? options = null,
+        CancellationToken cancellationToken = default)
+        => SetAsync<AutoCompleteDocument<Guid>, Guid>(document, options, cancellationToken);
+
+    public virtual Task<SetResponse> SetMultiAsync(IEnumerable<AutoCompleteDocument<Guid>> documents, SetOptions? options = null,
```

Comment:
SetMulti? SetList? or Set?

Reply:
In order to make single and multiple operations clearer, it is divided into two. It is more suitable to look at SetList here, but it is a bit strange to put it into delete operation. IRepository uses Range, but it doesn't feel good to use SetRange here.